### ISSUE TITLE
Optimization for Pi Zero W

### ISSUE THREAD

**Issue by romulusmare** [Sennevds/system_sensors#84]
Is there a way to optimize the python code?

Running the system service all works perfectly, but while updates are each 60 seconds, the htop result is like this:

`  PID USER      PRI  NI  VIRT   RES   SHR S CPU% MEM%   TIME+  Command
 1082 username   20   0 49244 20984 10848 S  0.0  4.8 33:21.13 /usr/bin/python3 /home/username/system_sensors/src/system_sensors.py /home/username/system_sensors/src/settings.yaml
  292 avahi      20   0  5884  2888  2552 S  0.0  0.7  0:10.84 avahi-daemon: running [raspberrypi.local]
    1 root       20   0 32660  7916  6352 S  0.0  1.8  0:09.78 /sbin/init
  462 username   20   0 12224  4072  3272 S  0.0  0.9  0:05.16 sshd: username@pts/0
   74 root       20   0 15792  6408  5576 S  0.0  1.5  0:01.76 /lib/systemd/systemd-journald
 3488 username   20   0  9200  4048  2400 R  3.1  0.9  0:01.36 htop
  111 root       20   0 17980  3872  3060 S  0.0  0.9  0:01.10 /lib/systemd/systemd-udevd
  463 username   20   0  8472  3784  2864 S  0.0  0.9  0:00.93 -bash
  221 systemd-t  20   0 22400  5556  4888 S  0.0  1.3  0:00.91 /lib/systemd/systemd-timesyncd
  256 messagebu  20   0  6532  3464  3008 S  0.0  0.8  0:00.75 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
  343 root       20   0 10988  4032  3440 S  0.0  0.9  0:00.56 wpa_supplicant -B -c/etc/wpa_supplicant/wpa_supplicant.conf -iwlan0 -Dnl80211,wext
  269 nobody     20   0  4304  2036  1860 S  0.0  0.5  0:00.55 /usr/sbin/thd --triggers /etc/triggerhappy/triggers.d/ --socket /run/thd.socket --user nobody --deviceglob /dev/input/event*
  251 root       20   0 13028  5768  5128 S  0.0  1.3  0:00.54 /lib/systemd/systemd-logind
  280 root       20   0 10728  4720  4316 S  0.0  1.1  0:00.44 /sbin/wpa_supplicant -u -s -O /run/wpa_supplicant
  448 username   20   0 14584  7284  6388 S  0.0  1.7  0:00.40 /lib/systemd/systemd --user
  275 root       20   0 27640  1440  1300 S  0.0  0.3  0:00.38 /usr/sbin/rngd -r /dev/hwrng
  430 root       20   0  2816  1704  1360 S  0.0  0.4  0:00.30 /sbin/dhcpcd -q -w
  295 root       20   0 24604  3548  2784 S  0.0  0.8  0:00.27 /usr/sbin/rsyslogd -n -iNONE
  445 root       20   0 12224  6332  5536 S  0.0  1.4  0:00.25 sshd: username [priv]
  305 root       20   0  7936  2096  1916 S  0.0  0.5  0:00.15 /usr/sbin/cron -f
  361 root       20   0  9520  4392  4020 S  0.0  1.0  0:00.14 /usr/lib/bluetooth/bluetoothd
  436 root       20   0 10712  5592  5016 S  0.0  1.3  0:00.10 /usr/sbin/sshd -D
  435 root       20   0  4292  1632  1528 S  0.0  0.4  0:00.03 /sbin/agetty -o -p -- \u --noclear tty1 linux
  306 avahi      20   0  5752  1436  1188 S  0.0  0.3  0:00.00 avahi-daemon: chroot helper
  357 root       20   0  2124   604   476 S  0.0  0.1  0:00.00 /usr/bin/hciattach /dev/serial1 bcm43xx 3000000 flow - b8:27:eb:48:e2:8b
  449 username   20   0 34200  3380  1744 S  0.0  0.8  0:00.00 (sd-pam)

`

Basically running this on a Raspberry Pi Zero W, CPU does not go below 12% at all time and htop sorted by TIME+ shows, the python script is basically working the CPU hard. I would expect this script to take very little processor power, only once every minute.

Total time for pythonscript is: 33:21.13. Next on sorted CPU TIME+ list is: 0:10.84.

**Comment by Sennevds** [comment]
If you have any propositions on minimizing this you can always create a pr. I made this for myself and thought this could be helpful for others but I'm not a python developer so don't know enough to minimize this.

**Comment by romulusmare** [comment]
I am not either, but in fact a long time C/C++ dev, Trouble is not sure I understood in short time I had how MQTT works, but I will give it a go. If you have anything to help me, feel free...

**Comment by Sennevds** [comment]
I'm a c# dev and just fiddle with python on the side

**Comment by Eddy805** [comment]
Are you sure it is this script that is causing the issue? I am also running on a Pi Zero W, and I am at around 4%, with htop itself using about 3%... And that's also running a local mosquitto instance for secure bridging.

Maybe if something is wrong with the `hass/status` subscription, the paho library starts using more CPU? I tried disabling the subscription (`client.subscribe` and the `on_message` stuff), but I don't see much change -- but in my case, there wasn't much to see anyway.
